From 3742fae4f424c67f952c8ddaefb084b3f86ade0b Mon Sep 17 00:00:00 2001
From: Jukka Laitinen <jukka.laitinen@intel.com>
Date: Wed, 24 May 2017 12:44:23 +0300
Subject: [PATCH] mfd: arizona: Request interrupt gpio for arizona to fix
 pinmux issues

In case the arizona IRQ is defined in the ACPI, request the GPIO to
make sure that the GPIO is muxed as input.

Signed-off-by: Jukka Laitinen <jukka.laitinen@intel.com>
---
 drivers/mfd/arizona-irq.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/drivers/mfd/arizona-irq.c b/drivers/mfd/arizona-irq.c
index 5e18d3c..519613a 100644
--- a/drivers/mfd/arizona-irq.c
+++ b/drivers/mfd/arizona-irq.c
@@ -207,6 +207,7 @@ int arizona_irq_init(struct arizona *arizona)
 	int ret, i;
 	const struct regmap_irq_chip *aod, *irq;
 	struct irq_data *irq_data;
+	struct gpio_desc *gpiod;
 
 	arizona->ctrlif_error = true;
 
@@ -339,7 +340,16 @@ int arizona_irq_init(struct arizona *arizona)
 	}
 
 	/* Used to emulate edge trigger and to work around broken pinmux */
-	if (arizona->pdata.irq_gpio) {
+
+	/* If the irq is not defined by pdata, but is defined in ACPI, set the
+	   gpio number into the pdata to fix the pinmux */
+	if (!arizona->pdata.irq_gpio) {
+		gpiod=devm_gpiod_get_index_optional(arizona->dev,
+						    "arizona IRQ", 0,
+						    GPIOD_IN);
+		if (gpiod)
+			arizona->pdata.irq_gpio=desc_to_gpio(gpiod);
+	} else {
 		if (gpio_to_irq(arizona->pdata.irq_gpio) != arizona->irq) {
 			dev_warn(arizona->dev, "IRQ %d is not GPIO %d (%d)\n",
 				 arizona->irq, arizona->pdata.irq_gpio,
-- 
2.7.4

